Nodejs/Product/Npm/SPI/NodeModules.cs

```diff
@@ -34,12 +34,15 @@ internal class NodeModules : AbstractNodeModules {
                 foreach (var dependency in parentPackageJson.AllDependencies) {
                     Package module = null;
                     if (!Contains(dependency.Name)) {
-                        module = new Package(
-                            parent,
-                            Path.Combine(modulesBase, dependency.Name),
-                            showMissingDevOptionalSubPackages);
-                        if (parent as IPackage == null || !module.IsMissing || showMissingDevOptionalSubPackages) {
-                            AddModule(module);
+                        var dependencyPath = Path.Combine(modulesBase, dependency.Name);
+                        if (dependencyPath.Length <= 224) {

```

Comment:
Yeah, digging into it would be best. For instance, right now this fix won't cover the case where you have a global package exceeding max_path.

Essentially you'l just want to perform the check on fullPathToRootDirectory